File: easy/major_element.pl
use strict;
use integer;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my @s = split( /,/, $line );
    my $n = scalar @s / 2;
    my %t;
    map { $t{$_}++; } @s;
    my $f = 0;
    foreach my $i ( keys %t ) {
        next if $t{$i} < $n;
        last if $t{$i} == $n;
        print "$i\n";
        $f = 1;
        last;
    }
    print "None\n" if !$f;
}
close(INFILE);

File: easy/major_element.rb
File.open(ARGV[0]).each_line do |line|
  found = false
  s = line.split(',').map(&:to_i)
  s.uniq.each do |i|
    next if s.count(i) < s.length / 2
    break if s.count(i) == s.length / 2
    puts i
    found = true
    break
  end
  puts 'None' unless found
end
